app/src/main/java/org/traccar/client/StatusWidget.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 - 2022 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.client
+
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.*
+import android.widget.RemoteViews
+
+import androidx.preference.PreferenceManager
+
+/**
+ * Implementation of Status Widget functionality, which can display the status of the service -
+ * whether it is enabled or not.
+ */
+class StatusWidget : AppWidgetProvider() {
+
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        update(context, appWidgetManager, appWidgetIds)
+    }
+
+    override fun onEnabled(context: Context) {
+        // When the first widget is created, we want to start receiving service status broadcasts,
+        // so we can react and change the look
+        val filter = IntentFilter()
+        filter.addAction(TrackingService.ACTION_STARTED)
+        filter.addAction(TrackingService.ACTION_STOPPED)
+        // Note: we must register through the app context as a workaround
+        // for 'BroadcastReceiver components are not allowed to register to receive intents'
+        context.applicationContext.registerReceiver(Companion, filter)
+    }
+
+    override fun onDisabled(context: Context) {
+        // When the last widget is disabled (removed), we stop receiving service status broadcasts
+        context.applicationContext.unregisterReceiver(Companion)
+    }
+
+    companion object: BroadcastReceiver() {
+
+        override fun onReceive(context: Context, intent: Intent) {
+            if (TrackingService.ACTION_STARTED == intent.action
+                    || TrackingService.ACTION_STOPPED == intent.action) {
+                updateWidgets(context)
+            }
+        }
+
+        // Performs update for all widgets of this class.
```

Comment:
Can we please remove all or most the comments. They seem pretty obvious. We only use comments when something is not trivial to understand based on the code.

Reply:
I believe, the comments (even trivial and obvious) are useful in open source project, so the code is easier to understand for contributors (especially for first-time ones).
Anyway I will remove these comments, if you consider them useless.

Reply:
I think the code should be self-documenting in most cases. There are very rare cases where comments are needed.